Add support for CMPv2 configuration (#2330)
hashicorp/terraform-provider-vault: website/docs/d/pki_secret_backend_config_cmpv2.html.md

---

# vault\_pki\_secret\_backend\_config\_cmp

Reads the PKI CMPv2 configuration from Vault Enterprise.

~> **Important** All data retrieved from Vault will be
written in cleartext to state file generated by Terraform, will appear in
the console output when Terraform runs, and may be included in plan files
if secrets are interpolated into any resource attributes.
Protect these artifacts accordingly. See
[the main provider documentation](../index.html)
for more details.

## Example Usage

```hcl
resource "vault_mount" "pki" {
  path        = "pki"
  type        = "pki"
  description = "PKI secret engine mount"
}

data "vault_pki_secret_backend_config_cmpv2" "cmpv2_config" {
  backend     = vault_mount.pki.path
}
```

## Argument Reference

The following arguments are supported:

* `namespace` - (Optional) The namespace of the target resource.
  The value should not contain leading or trailing forward slashes.
  The `namespace` is always relative to the provider's configured [namespace](/docs/providers/vault/index.html#namespace).
  *Available only for Vault Enterprise*.

* `backend` - (Required) The path to the PKI secret backend to

>   read the EST configuration from, with no leading or trailing `/`s.

Reviewer: ```suggestion
  read the CMPv2 configuration from, with no leading or trailing `/`s.
```

# Attributes Reference

> * `authenticators` - Lists the mount accessors EST should delegate authentication requests towards (see [below for nested schema](#nestedatt--authenticators)).

Reviewer: ```suggestion
* `authenticators` - Lists the mount accessors CMPv2 should delegate authentication requests towards (see [below for nested schema](#nestedatt--authenticators)).
```

> * `default_path_policy` - Required to be set if default_mount is enabled. Specifies the behavior for requests using the default EST label. Can be sign-verbatim or a role given by role:<role_name>.

Reviewer: ```suggestion
* `default_path_policy` - Specifies the behavior for requests using the non-role-qualified CMPv2 requests. Can be sign-verbatim or a role given by role:<role_name>.
```

* `enable_sentinel_parsing` - If set, parse out fields from the provided CSR making them available for Sentinel policies.

> * `enabled` - Specifies whether EST is enabled.

Reviewer: ```suggestion
* `enabled` - Specifies whether CMPv2 is enabled.
```
